plugins/lookup/hashi_vault.py

```diff
@@ -1,109 +1,229 @@
-  author: Jonathan Davila <jdavila(at)ansible.com>
-  short_description: retrieve secrets from HashiCorp's vault
+  author:
+    - Jonathan Davila <jdavila(at)ansible.com>
+    - Brian Scholer (@briantist)
+  short_description: Retrieve secrets from HashiCorp's vault
   requirements:
     - hvac (python library)
+    - hvac 0.7.0+ (for namespace support)
+    - hvac 0.9.6+ (to avoid all deprecation warnings)
+    - botocore (only if inferring aws params from boto)
+    - boto3 (only if using a boto profile)
   description:
-    - retrieve secrets from HashiCorp's vault
+    - Retrieve secrets from HashiCorp's vault.
   notes:
     - Due to a current limitation in the HVAC library there won't necessarily be an error if a bad endpoint is specified.
-    - As of Ansible 2.10, only the latest secret is returned when specifying a KV v2 path.
+    - As of Ansible 2.10, only the latest version of a secret is returned when specifying a KV v2 path.
+    - As of Ansible 2.10, all options can be supplied via term string (space delimited key=value pairs) or by parameters (see examples).
+    - As of Ansible 2.10, when C(secret) is the first option in the term string, C(secret=) is not required (see examples).
   options:
     secret:
       description: query you are making.
       required: True
     token:
-      description: vault token.
+      description:
+        - Vault token. If using token auth and no token is supplied, explicitly or through env, then the plugin will check
+        - for a token file, as determined by C(token_path) and C(token_file).
       env:
         - name: VAULT_TOKEN
+    token_path:
+      description: If no token is specified, will try to read the token file from this path.
+      env:
+        - name: HOME
+      ini:
+        - section: lookup_hashi_vault
+          key: token_path
+    token_file:
+      description: If no token is specified, will try to read the token from this file in C(token_path).
+      ini:
+        - section: lookup_hashi_vault
+          key: token_file
+      default: '.vault-token'
     url:
       description: URL to vault service.
       env:
         - name: VAULT_ADDR
+      ini:
+        - section: lookup_hashi_vault
+          key: url
       default: 'http://127.0.0.1:8200'
     username:
       description: Authentication user name.
     password:
       description: Authentication password.
     role_id:
-      description: Role id for a vault AppRole auth.
+      description: Vault Role ID. Used in approle and aws_iam_login auth methods.
       env:
         - name: VAULT_ROLE_ID
+      ini:
+        - section: lookup_hashi_vault
+          key: role_id
     secret_id:
       description: Secret id for a vault AppRole auth.
       env:
         - name: VAULT_SECRET_ID
     auth_method:
       description:
-      - Authentication method to be used.
-      - C(userpass) is added in version 2.8.
+        - Authentication method to be used.
+        - C(userpass) is added in version 2.8.
```

Comment:
```suggestion
        - C(userpass) is added in Ansible 2.8.
```
To make it clear that you're not talking about the community.general version.

Reply:
Done!